pkg/control_room/control_center.go

```diff
@@ -0,0 +1,92 @@
+package control_room
+
+import (
+	"container/heap"
+
+	"github.com/TNSEngineerEdition/WailsClient/pkg/city"
+	"github.com/umahmood/haversine"
+)
+
+type GraphNode = city.GraphNode
+
+type ControlCenter struct {
+	city         *city.City
+	cachedRoutes map[[2]uint64][]GraphNode
+}
+
+func CreateControlCenter(city *city.City) *ControlCenter {
+	return &ControlCenter{
+		city:         city,
+		cachedRoutes: make(map[[2]uint64][]GraphNode),
+	}
+}
+
+func (cc *ControlCenter) GetShortestPath(sourceID, destID uint64) []GraphNode {
```

Comment:
The convention is to use one-letter names for the current struct in methods.
```suggestion
func (c *ControlCenter) GetShortestPath(sourceID, destID uint64) []*GraphNode {
```